lexicon/providers/dreamhost.py

```
@@ -0,0 +1,181 @@
+    def _list_records(self, rtype=None, name=None, content=None):
+
+        payload = self._get('dns-list_records')
+
+        if payload.get('result', '') != 'success':
+            raise Exception('unable to get records: %s' % payload)
+
+        resource_list = payload.get('data', None)
+        if resource_list is None:
+            raise Exception('unable to get records: %s' % payload)
+
+        resource_list = [
+            resource for resource in resource_list if resource['zone'] == self.domain]
+        if rtype:
+            resource_list = [
+                resource for resource in resource_list if resource['type'] == rtype]
+        if name:
+            name = self._full_name(name)
+            resource_list = [
+                resource for resource in resource_list if resource['record'] == name]
+        if content:
+            resource_list = [
+                resource for resource in resource_list if resource['value'] == content]
+
+        processed_records = []
+        for resource in resource_list:
+            processed_records.append({
+                'id': resource['type'] + '|' + resource['record'] + '|' + resource['value'],
+                'type': resource['type'],
+                'name': resource['record'],
+                'content': resource['value'],
+            })
+
+        return processed_records
+
+    # Create or update a record.
+    def _update_record(self, identifier, rtype=None, name=None, content=None):
+        if identifier == '':
+            identifier = None
+        if identifier is not None:
+            id_list = identifier.split('|')
+            if len(id_list) != 3:
+                raise Exception('invalid identifier: %s' % (identifier))
+
+            id_type, id_name, id_content = id_list
+
+            try:
+                self._delete_record(rtype=id_type, name=id_name, content=id_content)
+            except:  # pylint: disable=bare-except
+                pass
```

Can you put a comment here, to explain what is the expected exception, and why it is not a problem ?

---

This is "trying to delete old record" in _update_record.
It's possible that the exception is because record is already deleted / does not exist.

I thought the focus of _update_record is to create the expected record,
and it's ok to ignore the exception in _delete_record.

Of course we can fine-tune this part to restrict except to only "non-exist".

---

In this case yes I think we should focus on ignoring the request only if an http error occurred because of non-existant record. The other unexpected exceptions should stop the process right here.